packages/@uppy/companion/src/server/middlewares.js

```diff
@@ -79,25 +79,22 @@ exports.loadSearchProviderToken = (req, res, next) => {
 }
 
 exports.cors = (options = {}) => (req, res, next) => {
-  const exposedHeaders = [
-    // exposed so it can be accessed for our custom uppy client preflight
-    'Access-Control-Allow-Headers',
-  ]
-  if (options.sendSelfEndpoint) exposedHeaders.push('i-am')
-  if (res.get('Access-Control-Expose-Headers')) exposedHeaders.push(res.get('Access-Control-Expose-Headers'))
-
-  const allowedHeaders = [
-    'uppy-auth-token',
-    'uppy-versions',
-    'uppy-credentials-params',
-  ]
-  if (res.get('Access-Control-Allow-Headers')) allowedHeaders.push(res.get('Access-Control-Allow-Headers'))
-
   // TODO: Move to optional chaining when we drop Node.js v12.x support
-  const ACAMHeader = res.get('Access-Control-Allow-Methods')
-  const existingAllowMethodsHeader = new Set(ACAMHeader && ACAMHeader.split(',').map(method => method.trim().toUpperCase()))
-  existingAllowMethodsHeader.add('GET').add('POST').add('OPTIONS').add('DELETE')
-  const methods = Array.from(existingAllowMethodsHeader)
+  const existingExposeHeaders = res.get('Access-Control-Expose-Headers')
+  const exposeHeadersMap = new Set(existingExposeHeaders && existingExposeHeaders.split(',').map(method => method.trim().toLowerCase()))
+  // exposed so it can be accessed for our custom uppy client preflight
+  exposeHeadersMap.add('access-control-allow-headers')
+  if (options.sendSelfEndpoint) exposeHeadersMap.add('i-am')
+
+  const existingAllowHeaders = res.get('Access-Control-Allow-Headers')
+  const allowHeadersMap = new Set(existingAllowHeaders && existingAllowHeaders.split(',').map(method => method.trim().toLowerCase()))
+  // Needed for basic operation: https://github.com/transloadit/uppy/issues/3021
+  allowHeadersMap.add('uppy-auth-token').add('uppy-versions').add('uppy-credentials-params').add('authorization').add('origin').add('content-type').add('accept')
```

Comment:
nit: instead of a long chain of `.add`s, we could define an array and pass it to the constructor:
```suggestion
  // Needed for basic operation: https://github.com/transloadit/uppy/issues/3021
  const allowedHeaders = [
    'uppy-auth-token',
    'uppy-versions',
    'uppy-credentials-params',
    'authorization',
    'origin',
    'content-type',
    'accept',
  ];
  const existingAllowHeaders = res.get('Access-Control-Allow-Headers');
  const allowHeadersMap = new Set(existingAllowHeaders ?
    existingAllowHeaders
      .split(',')
      .map((method) => method.trim().toLowerCase())
      .concat(allowedHeaders)
    : allowedHeaders
  );
```